### PR TITLE
ROC-5170 Company contact name

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5170

### Change description

Add page for company contact details

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [x] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/mediation/routes/can-we-use-company.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+}
+
+function getContactName (res: express.Response) {
+  const claim: Claim = res.locals.claim
+  if (!claim.isResponseSubmitted()) {
+    const draftResponse: Draft<ResponseDraft> = res.locals.responseDraft
+    return (draftResponse.document.defendantDetails.partyDetails as CompanyDetails).contactPerson ? (draftResponse.document.defendantDetails.partyDetails as CompanyDetails).contactPerson : undefined
+  } else {
+    return (claim.claimData.claimant as CompanyDetails).contactPerson
```
Comment (KiranVarma02) [inline src/main/features/mediation/routes/can-we-use-company.ts]: why `as` ? I am curious to know ..

File: src/main/features/response/tasks/freeMediationTask.ts [hunk 2]
```diff
@@ -14,7 +14,10 @@ export class FreeMediationTask {
         !!mediationDraft.willYouTryMediation && mediationDraft.willYouTryMediation.option === FreeMediationOption.YES &&
         !!mediationDraft.youCanOnlyUseMediation &&
         (mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.YES ||
-          mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.NO)
+          mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.NO) &&
+        ((!mediationDraft.canWeUse && !mediationDraft.canWeUseCompany) ||
+          (!!mediationDraft.canWeUse && mediationDraft.canWeUse.isCompleted()) ||
+          (!!mediationDraft.canWeUseCompany && mediationDraft.canWeUseCompany.isCompleted()))
```
Comment (KiranVarma02) [inline src/main/features/response/tasks/freeMediationTask.ts]: looks scary but its actually not that bad i guess...

File: src/main/features/response/tasks/freeMediationTask.ts [hunk 3]
```diff
@@ -14,7 +14,10 @@ export class FreeMediationTask {
         !!mediationDraft.willYouTryMediation && mediationDraft.willYouTryMediation.option === FreeMediationOption.YES &&
         !!mediationDraft.youCanOnlyUseMediation &&
         (mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.YES ||
-          mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.NO)
+          mediationDraft.youCanOnlyUseMediation.option === FreeMediationOption.NO) &&
+        ((!mediationDraft.canWeUse && !mediationDraft.canWeUseCompany) ||
```
Comment (singh-surjeet) [inline src/main/features/response/tasks/freeMediationTask.ts]: I agree - can this be simplified?

File: src/test/http-mocks/claim-store.ts [hunk 4]
```diff
@@ -22,10 +22,64 @@ import {
 } from 'test/data/entity/responseData'
 import { PaymentOption } from 'claims/models/paymentOption'
 import { PaymentSchedule } from 'claims/models/response/core/paymentSchedule'
+import { organisation } from '../data/entity/party'
```
Comment (singh-surjeet) [inline src/test/http-mocks/claim-store.ts]: Pls replace relative paths

File: src/main/features/response/tasks/freeMediationTask.ts (hunk 3 above)
Comment (robertparkinson) [reply src/main/features/response/tasks/freeMediationTask.ts]: done.

File: src/test/http-mocks/claim-store.ts (hunk 4 above)
Comment (robertparkinson) [reply src/test/http-mocks/claim-store.ts]: done

**Pull request merged**